packages/eas-cli/src/utils/code-signing.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  convertCertificatePEMToCertificate,
+  convertPrivateKeyPEMToPrivateKey,
+  signStringRSASHA256AndVerify,
+  validateSelfSignedCertificate,
+} from '@expo/code-signing-certificates';
+import Dicer from 'dicer';
+import isDeepEqual from 'fast-deep-equal';
+import { promises as fs } from 'fs';
+import { Response } from 'got';
+import { pki as PKI } from 'node-forge';
+import nullthrows from 'nullthrows';
+import { Stream } from 'stream';
+import { parseItem } from 'structured-headers';
+
+import { PartialManifest, PartialManifestAsset } from '../graphql/generated';
+
+export async function getKeyAndCertificateFromPathsAsync({
+  codeSigningCertificatePath,
+  privateKeyPath,
+}: {
+  codeSigningCertificatePath: string;
+  privateKeyPath: string;
+}): Promise<{ privateKey: PKI.rsa.PrivateKey; certificate: PKI.Certificate }> {
+  const [codeSigningCertificatePEM, privateKeyPEM] = await Promise.all([
+    fs.readFile(codeSigningCertificatePath, 'utf8'),
+    fs.readFile(privateKeyPath, 'utf8'),
+  ]);
+
+  const privateKey = convertPrivateKeyPEMToPrivateKey(privateKeyPEM);
+  const certificate = convertCertificatePEMToCertificate(codeSigningCertificatePEM);
+  validateSelfSignedCertificate(certificate, {
+    publicKey: certificate.publicKey as PKI.rsa.PublicKey,
+    privateKey,
+  });
+
+  return {
+    privateKey,
+    certificate,
+  };
+}
+
+export type MultipartPart = { headers: Map<string, string>; body: string };
+
+export async function parseMultipartMixedResponseAsync(res: Response): Promise<MultipartPart[]> {
+  const contentType = res.headers['content-type'];
+  if (!contentType) {
+    throw new Error('Missing content-type in multipart response');
+  }
+
+  const boundaryRegex = /^multipart\/.+?; boundary=(?:"(.+)"|([^\s]+))$/i;
+  const matches = boundaryRegex.exec(contentType);
+  if (!matches) {
+    throw new Error('content-type header in response not multipart');
+  }
+  const boundary = matches[1] ?? matches[2];
```

Comment:
what is the situation in which `matches[1]` would be undefined, but `matches[2]` would be the correct?

Reply:
The boundary is not necessarily always in quotes, so `matches[1]` will be the case when it is quoted (probably contains whitespace) and `matches[2]` will be the case otherwise (can't contain whitespace). Now that I think about it, we may be able to use structured-headers to parse this?

Reply:
Looks parseable https://runkit.com/ide/structured-headers-test

Reply:
Just tried, seems like our parser can't parse it. Will create a task for a follow-up investigation.